### DISCUSSION TITLE
refreshNuxtData fetching multiple time

### DISCUSSION THREAD

**Discussion by irsyaadbp** [nuxt/nuxt#25472] [General]
Hello,
i try to revalidate nuxt data with refreshNuxtData with specific key
but in network calling twice

the simple code is

```js
function useHttp(endpoint, options) {
  const methods = options.methods || 'GET'
  const key = options?.key || `${endpoint}-${methods}`

  const defaultOptions = {
    key,
    async onResponseError() {
      if (errorCode === 401) {
        await refreshToken()
        await refreshNuxtData(key)
      }
    },
  }

  const config = defu(options, defaultOptions)

  return useFetch(endpoint, config)
}
```

and i call like this
```js
const { data } = await useHttp('/api/users')
```

if 401 in first time and second working like expected, but in third, fourth, and subsequent ones will perform multiplicative refetching.

and i already tried with nuxt devtools too
in devtools only has 1 data with the key, and i refresh from nuxt devtools also calling twice

**Comment by leindfraust** [comment]
Having this the same problem
